### PR TITLE
fix: Remove hard coded strings when showing the showShareSheet

### PR DESCRIPTION
**Pull request by krysalead**
This change break the backward compatibility because the third parameter of showShareSheet action is now a JSON and not a String.

### CODE REVIEW

**Comment by aaustin** [comment]
Thanks for expanding the API! I'd love to send you a sweatshirt: https://branch.io/branch-referral-program-claim-reward/

I can't edit your PR, but would you mind just adding a conditional which checks if third param is type String vs JSON, then falling back to the old method if so? Along with that, please confirm that it won't break if you try to pull additional parameters while the dev is still using the legacy interface. 

That would help!

**Comment by krysalead** [comment]
Hi,

According to your comment I have updated the Java code to handle both JSON and string. I also update the Readme as it was not proper.
I had hard time to work in your environment with the "semantic-release" tool your are using as it is hard to install on Linux at least.

File: src/android/io/branch/BranchSDK.java [hunk 1]
```diff
@@ -1165,7 +1130,20 @@ public void run() {
...
                         registerView(this.args.getInt(0), this.callbackContext);
                     } else if (this.action.equals("showShareSheet")) {
-                        showShareSheet(this.args.getInt(0), this.args.getJSONObject(1), this.args.getJSONObject(2), this.args.getString(3) != null ? this.args.getString(3) : "This stuff is awesome: ");
+                        Object userLocalization = this.args.opt(3);
+                        JSONObject localization = new JSONObject();
+                        //Limitation is that the user must override all the strings
+                        if (userLocalization != null && userLocalization instanceof JSONObject) {
+                            localization = (JSONObject) userLocalization;
+                        } else {
+                            localization.put("shareText", "This stuff is awesome:");
```
Comment (aaustin) [inline src/android/io/branch/BranchSDK.java]: Would you mind changing this to:
```
localization.put("shareText", this.args.getString(3) != null ? this.args.getString(3) : "This stuff is awesome: ");
```

Then I think we'll be all set to merge this guy in.

Comment (krysalead) [inline src/android/io/branch/BranchSDK.java]: yep

**Comment by krysalead** [comment]
Hi Alex,

I filled the form to receive my swag :) is it still a valid proposal?

thanks & regards

Olivier

**Comment by aaustin** [comment]
Of course ! You should be receiving a confirmation in the next week :)

**Comment by mada299** [comment]
Olivier,

Can you send me - mada [at] branch.io your name, address and hoodie size you'd like?
Thanks again :)

--Mada

**Pull request merged**